diff --git a/modules/pmg_tk/PMGApp.py b/modules/pmg_tk/PMGApp.py
index a3f36e37..29bf23db 100644
--- a/modules/pmg_tk/PMGApp.py
+++ b/modules/pmg_tk/PMGApp.py
@@ -346,11 +346,19 @@ class PMGApp(Pmw.MegaWidget):
 
             inv = sys.modules.get("pymol.invocation",None)
             if inv != None:
+                # for aqua the window needs to be a little bigger
+                if self.root._windowingsystem == 'aqua':
+                    widthPad = 400
+                    heightPad = 50
+                else:
+                    widthPad = 220
+                    heightPad = 0
+
                 if skin == None:
                     skin = inv.options.skin
-                self.frameWidth = inv.options.win_x + 220
+                self.frameWidth = inv.options.win_x + widthPad
                 self.frameXPos = inv.options.win_px - self.frameXAdjust
-                self.frameHeight = inv.options.ext_y
+                self.frameHeight = inv.options.ext_y + heightPad
                 self.frameYPos = inv.options.win_py - (
                          self.frameHeight + self.frameYAdjust)
                 self.setSkin(skin,run=0)
diff --git a/modules/pmg_tk/skins/normal/__init__.py b/modules/pmg_tk/skins/normal/__init__.py
index e919216a..9972f248 100644
--- a/modules/pmg_tk/skins/normal/__init__.py
+++ b/modules/pmg_tk/skins/normal/__init__.py
@@ -422,7 +422,9 @@ class Normal(PMGSkin):
         self.command.set(self.history[self.history_cur])
         l = len(self.history[self.history_cur])
         self.entry.icursor(l)
-    
+
+        return 'break'
+
     def forward(self):
         if not self.history_cur:
             self.history[0] = self.command.get()
@@ -431,6 +433,8 @@ class Normal(PMGSkin):
         l = len(self.history[self.history_cur])
         self.entry.icursor(l)
 
+        return 'break'
+
     def doAsync(self,cmmd):
         t = threading.Thread(target=_doAsync,args=(self.cmd,cmmd))
         t.setDaemon(1)
diff --git a/modules/pymol/__init__.py b/modules/pymol/__init__.py
index 90b645b4..655e6fe3 100644
--- a/modules/pymol/__init__.py
+++ b/modules/pymol/__init__.py
@@ -414,6 +414,21 @@ def prime_pymol():
             # launch X11 (if needed)
             os.system("/usr/bin/open -a X11")
 
+def _tk_is_aqua():
+    if sys.platform != 'darwin':
+        return False
+
+    if sys.version_info[0] == 2:
+        import Tkinter as tkinter
+    else:
+        import tkinter
+
+    root = tkinter.Tk()
+    try:
+        return root._windowingsystem == 'aqua'
+    finally:
+        root.destroy()
+
 def launch(args=None, block_input_hook=0):
     '''
     Run PyMOL with args
@@ -423,6 +438,17 @@ def launch(args=None, block_input_hook=0):
     if args is None:
         args = sys.argv
     invocation.parse_args(args)
+
+    # hack for aqua Tk: swap threads
+    if invocation.options.gui == 'pmg_tk' \
+            and invocation.options.external_gui == 1 \
+            and _tk_is_aqua():
+        import pmg_tk, pymol
+        finish_launching(['pymol', '-x'])  # async
+        cmd.reaper = None  # avoid hanging on "quit"
+        pmg_tk.run(pymol, 0)  # main thread
+        return
+
     prime_pymol()
     _cmd.runpymol(_cmd._get_global_C_object(), block_input_hook)
 
