diff --git a/modules/pmg_tk/PMGApp.py b/modules/pmg_tk/PMGApp.py
index a3f36e3..29bf23d 100644
--- a/modules/pmg_tk/PMGApp.py
+++ b/modules/pmg_tk/PMGApp.py
@@ -346,11 +346,19 @@ class PMGApp(Pmw.MegaWidget):
 
             inv = sys.modules.get("pymol.invocation",None)
             if inv != None:
+                # for aqua the window needs to be a little bigger
+                if self.root._windowingsystem == 'aqua':
+                    widthPad = 400
+                    heightPad = 50
+                else:
+                    widthPad = 220
+                    heightPad = 0
+
                 if skin == None:
                     skin = inv.options.skin
-                self.frameWidth = inv.options.win_x + 220
+                self.frameWidth = inv.options.win_x + widthPad
                 self.frameXPos = inv.options.win_px - self.frameXAdjust
-                self.frameHeight = inv.options.ext_y
+                self.frameHeight = inv.options.ext_y + heightPad
                 self.frameYPos = inv.options.win_py - (
                          self.frameHeight + self.frameYAdjust)
                 self.setSkin(skin,run=0)
diff --git a/modules/pmg_tk/skins/normal/__init__.py b/modules/pmg_tk/skins/normal/__init__.py
index a433134..78970c6 100644
--- a/modules/pmg_tk/skins/normal/__init__.py
+++ b/modules/pmg_tk/skins/normal/__init__.py
@@ -390,6 +390,7 @@ class Normal(PMGSkin, pymol._gui.PyMOLDesktopGUI):
 
     def command_set_cursor(self, i):
         self.entry.icursor(i)
+        return 'break'
 
     def dump(self,event):
         print(dir(event))
diff --git a/modules/pymol/__init__.py b/modules/pymol/__init__.py
index 6a62f63..2959330 100644
--- a/modules/pymol/__init__.py
+++ b/modules/pymol/__init__.py
@@ -97,6 +97,21 @@ import math
 
 from . import invocation
 
+def _tk_is_aqua():
+    if sys.platform != 'darwin':
+        return False
+
+    if sys.version_info[0] == 2:
+        import Tkinter as tkinter
+    else:
+        import tkinter
+
+    root = tkinter.Tk()
+    try:
+        return root._windowingsystem == 'aqua'
+    finally:
+        root.destroy()
+
 def _init_internals(_pymol):
 
     # Create a temporary object "stored" in the PyMOL global namespace
@@ -444,6 +459,16 @@ def launch(args=None, block_input_hook=0):
             print('Qt not available, using GLUT/Tk interface')
             invocation.options.gui = 'pmg_tk'
 
+    # hack for aqua Tk: swap threads
+    if invocation.options.gui == 'pmg_tk' \
+            and invocation.options.external_gui == 1 \
+            and _tk_is_aqua():
+        import pmg_tk, pymol
+        finish_launching(['pymol', '-x'])  # async
+        cmd.reaper = None  # avoid hanging on "quit"
+        pmg_tk.run(pymol, 0)  # main thread
+        return
+
     prime_pymol()
     _cmd.runpymol(_cmd._get_global_C_object(), block_input_hook)
 
